### ISSUE TITLE
Attachment link maybe fails 2.5.8 Target Size (Minimum)

### ISSUE THREAD

**Issue by selfthinker** [alphagov/govuk_publishing_components#3444]
The [Attachment link component](https://components.publishing.service.gov.uk/component-guide/attachment_link) maybe fails [2.5.8 Target Size (Minimum)](https://w3c.github.io/wcag/understanding/target-size-minimum.html).

Note: 2.5.8 is part of the new draft WCAG 2.2 criteria. The latest standard is WCAG 2.1. WCAG 2.2 isn't published yet, so this is potentially subject to change.

The Attachment link component has two cases in the component guide where the link is not big enough: [with target blank](https://components.publishing.service.gov.uk/component-guide/attachment_link#with_target_blank) and [with data attributes](https://components.publishing.service.gov.uk/component-guide/attachment_link#with_data_attributes).

The link plus its spacing needs to be at least 24x24px. But those links are only 20px high.
This is fine in other examples in the guide (like in 'Embedded in govspeak') because those are used inline with text.

Although this fails within the component guide, it might not fail when used within context on GOV.UK. As the component misses its context, I cannot fully evaluated it.

If the component is used with other text around it, it’s a pass.
If the component is used within another element which gives enough spacing before and/or after it (so that it’s together at least 24 pixels), it’s a pass.
Otherwise it’s a fail.

**Comment by jon-kirwan** [comment]
I think the only thing we need to do is update the component guide examples.

The component guide already includes guidance for the correct usage:
> It is intended to be embedded within an element that supplies text styles (like `.govuk-body`), so it doesn't include its own text styling

Searching GitHub, I find [2 occurrences](https://github.com/search?q=org%3Aalphagov+components%2Fcomponents%2Fattachment_link&type=Coded) (not including the one from this repo):
- Content publisher (here the component is embedded in a paragraph element)
- Govspeak

I looked at the following pages for examples embedded in Govspeak:
- https://www.gov.uk/government/statistical-data-sets/vehicle-licensing-statistics-data-tables
- https://www.gov.uk/government/statistical-data-sets/unclaimed-estates-list
- https://www.gov.uk/government/statistical-data-sets/public-sector-trade-union-facility-time-data
- https://www.gov.uk/guidance/oil-and-gas-eems-database
- https://www.gov.uk/guidance/criminal-justice-system-data-standards-forum-guidance 

In each example, the attachment link is usually placed within a paragraph or list item element, and the target (height) is at least 25px.

Referring to @selfthinker's comments:
> If the component is used within another element which gives enough spacing before and/or after it (so that it’s together at least 24 pixels), it’s a pass.

It's difficult to review more than a few pages, but it appears that publishers typically add attachments in one of three ways: they either use a list, insert double line breaks to create paragraph spacing, or embed the links within a paragraph of text.

So, I think, all we should do here is update the two component guide examples ([with target blank](https://components.publishing.service.gov.uk/component-guide/attachment_link#with_target_blank) and [with data attributes](https://components.publishing.service.gov.uk/component-guide/attachment_link#with_data_attributes)) to be embedded in a paragraph element to indicate the correct usage.
